R/pandoc.R

```diff
@@ -472,44 +472,77 @@ pandoc_self_contained_html <- function(input, output) {
 }
 
 
-validate_self_contained <- function(mathjax) {
+validate_self_contained <- function(math) {
 
-  if (identical(mathjax, "local"))
-    stop2("Local MathJax isn't compatible with self_contained\n",
-         "(you should set self_contained to FALSE)")
+  if (identical(c(names(math), math[[1L]]), c("mathjax", "local")))
+    stop("Local MathJax isn't compatible with self_contained\n",
+         "(you should set self_contained to FALSE)", call. = FALSE)
 }
 
-pandoc_mathjax_args <- function(mathjax,
-                                template,
-                                self_contained,
-                                files_dir,
-                                output_dir) {
```

Comment:
In general, packages should not directly depend on internal functions from other packages...
Anyway, we can start from deprecating the function.

Reply:
For reference in the context of **rmarkdown**, we always need to check that the RStudio IDE code does not use one internal function. It can happen - RStudio IDE use some internal functions of package sometimes. (until they implement their own version)

Reply:
It seems like it is not used. So we'll probably remove this function.